Sources/Zephyr.swift

```diff
@@ -447,12 +447,17 @@ extension Zephyr {
     ///  Observation method for `NSUbiquitousKeyValueStore.didChangeExternallyNotification`
     func keysDidChangeOnCloud(notification: Notification) {
         if notification.name == NSUbiquitousKeyValueStore.didChangeExternallyNotification {
-            guard let userInfo = (notification as NSNotification).userInfo,
-                let cloudKeys = userInfo[NSUbiquitousKeyValueStoreChangedKeysKey] as? [String],
-                let localStoredDate = zephyrLocalStoreDictionary[ZephyrSyncKey] as? Date,
-                let remoteStoredDate = zephyrRemoteStoreDictionary[ZephyrSyncKey] as? Date,
-                remoteStoredDate.timeIntervalSince1970 > localStoredDate.timeIntervalSince1970 else {
-                    return
+            let userInfo = (notification as NSNotification).userInfo
+            let cloudKeys = userInfo?[NSUbiquitousKeyValueStoreChangedKeysKey] as? [String]
+            let localStoredDate = zephyrLocalStoreDictionary[ZephyrSyncKey] as? Date
+            let remoteStoredDate = zephyrRemoteStoreDictionary[ZephyrSyncKey] as? Date
+            
+            if (userInfo == nil || remoteStoredDate == nil || cloudKeys == nil) {
+                return
+            }
+            
+            if (localStoredDate != nil && remoteStoredDate != nil && !(remoteStoredDate!.timeIntervalSince1970 > localStoredDate!.timeIntervalSince1970)) {
```

Comment:
these are force-unwrapped. Mind addressing that? Thank you!

Reply:
Absolutely. Have added extra check and avoid force-unwrapping.

Reply:
`localStoredDate!` is still force unwrapped :)

Reply:
My bad, damn typos... Anyway, now it should actually be fixed ;)

Reply:
Thanks - will check it out this weekend

Reply:
We have force-unwrapping here again, see comment below.